request-example.go

```diff
@@ -69,6 +75,48 @@ func (fs *root) Filewrite(r *Request) (io.WriterAt, error) {
 	return file.WriterAt()
 }
 
+func (fs *root) CheckFile(r *Request) (CheckFileResponse, error) {
+	response := CheckFileResponse{}
+	requestedAlgos := r.CheckFile.GetHashAlgoAsList()
+	var h hash.Hash
+	for _, algo := range requestedAlgos {
+		if algo == "md5" {
```

Comment:
This should be a `switch … case` especially since every condition is `algo == A_VALUE`.